pkg/extractor/extractor.go

```diff
@@ -131,22 +131,22 @@ func extractConfigurationK8sData(content []byte, yamlNode yaml.Node) Configurati
 		return configuration
 	}
 
-	if metadata := jsonObject["metadata"]; metadata != nil {
-		if metadataName := metadata.(map[string]interface{})["name"]; metadataName != nil {
-			configuration.MetadataName = metadataName.(string)
+	if metadata, ok := jsonObject["metadata"]; ok {
+		if metadataName, ok := metadata.(map[string]interface{})["name"].(string); ok {
+			configuration.MetadataName = metadataName
 		}
 
-		if annotations := metadata.(map[string]interface{})["annotations"]; annotations != nil {
-			configuration.Annotations = annotations.(map[string]interface{})
+		if annotations, ok := metadata.(map[string]interface{})["annotations"].(map[string]interface{}); ok {
+			configuration.Annotations = annotations
 		}
 	}
 
-	if apiVersion := jsonObject["apiVersion"]; apiVersion != nil {
-		configuration.ApiVersion = apiVersion.(string)
+	if apiVersion, ok := jsonObject["apiVersion"].(string); ok {
+		configuration.ApiVersion = apiVersion
 	}
 
-	if kind := jsonObject["kind"]; kind != nil {
-		configuration.Kind = kind.(string)
+	if kind, ok := jsonObject["kind"].(string); ok {
```

Comment:
it means that we don't need the check of `kind != nil`? because of the use of `ok`?

Reply:
yeah it's fine :)